components/core/src/service.rs

```diff
@@ -16,7 +16,8 @@ lazy_static::lazy_static! {
         Regex::new(r"(?x)\A(
         (?P<application_environment>[^\#@]+)
         \#)?
-        (?P<service>[^\#@]+)
+        # allow periods in service name but not at start or end
+        (?P<service>[A-Za-z0-9_-]+([A-Za-z0-9_.-]+[A-Za-z0-9_-]+)?)
```

Comment:
`(?P<service>[A-Za-z0-9_-]+(?:[A-Za-z0-9_.-]+[A-Za-z0-9_-]+)?)`

I think we want to make that a non-capturing group, for clarity (the `?:` bit).